ksqldb-functional-tests/src/test/resources/query-validation-tests/greatest.json

```diff
@@ -0,0 +1,178 @@
+        {"topic": "OUTPUT", "key": 3, "value": {"HIGHEST": 500000.0}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST DOUBLE"}
+        ]
+      }
+    },
+    {
+      "name": "test decimal widening",
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY, N1 DECIMAL(9,2), N2 DECIMAL (7,6)) WITH (kafka_topic='input_topic',value_format='json');",
+        "CREATE STREAM OUTPUT AS SELECT ID, GREATEST(N1, N2) AS HIGHEST FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": 1, "value": {"N1": 1234567.89, "N2": 0.123456}, "N4": 0.0}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"HIGHEST": 1234567.89}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID BIGINT KEY, HIGHEST DECIMAL(13,6)"}
+        ]
+      }
+    },
+    {
+      "name": "test greatest with strings",
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY, S1 STRING, S2 STRING, S3 STRING, S4 STRING) WITH (kafka_topic='input_topic',value_format='json');",
+        "CREATE STREAM OUTPUT AS SELECT ID, GREATEST(S1, S2, S3, S4, null, null, 'hello') AS HIGHEST FROM INPUT;",
+        "CREATE STREAM OUTPUT2 AS SELECT ID, GREATEST('\"', S1) AS HIGHEST FROM INPUT;",
+        "CREATE STREAM OUTPUT3 AS SELECT ID, GREATEST('\\foo\"', S1) AS HIGHEST FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": 1, "value": {"S1": "apple", "S2": "banana", "S3": "aardvark",  "S4": null}},
+        {"topic": "input_topic", "key": 2, "value": {"S1": null, "S2": null, "S3": null,  "S4": null}},
+        {"topic": "input_topic", "key": 3, "value": {"S1": "!", "S2": "zebra", "S3": "aardvark",  "S4": null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 1, "value": {"HIGHEST": "hello"}},
+        {"topic": "OUTPUT", "key": 2, "value": {"HIGHEST": "hello"}},
+        {"topic": "OUTPUT", "key": 3, "value": {"HIGHEST": "zebra"}},
+
+        {"topic": "OUTPUT2", "key": 1, "value": {"HIGHEST":"apple"}},
+        {"topic": "OUTPUT2", "key": 2, "value": {"HIGHEST":"\""}},
+        {"topic": "OUTPUT2", "key": 3, "value": {"HIGHEST":"\""}},
+
+        {"topic": "OUTPUT3", "key": 1, "value": {"HIGHEST":"apple"}},
+        {"topic": "OUTPUT3", "key": 2, "value": {"HIGHEST":"\\foo\""}},
+        {"topic": "OUTPUT3", "key": 3, "value": {"HIGHEST":"\\foo\""}}
+      ]
+    },
+    {
+      "name": "test all null input",
+      "statements": [
+        "CREATE STREAM INPUT (ID BIGINT KEY) WITH (kafka_topic='input_topic',value_format='json');",
+
+        "CREATE STREAM OUTPUT1 AS SELECT ID, GREATEST(null, null, null, null) AS HIGHEST FROM INPUT;"
+      ],
+        "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+          "message": "Function 'greatest' does not accept parameters (null, null, null, null)"
```

Comment:
As before, I thought we were expecting a more specific/informative error message?